### DISCUSSION TITLE
[help] Best way to extend options in `targets`?

### DISCUSSION THREAD

**Discussion by Aariq** [ropensci/targets#1263] [Help] [answered]
### Help

- [X] I understand and agree to <https://books.ropensci.org/targets/help.html>.

### Description

With `geotargets`, we'd like to be able to set options that ultimately get passed to write functions and we'd ideally like to use the same API as targets with `tar_options_set()` and `tar_options_get()` just for consistency and familiarity for users. In playing around, I've realized that options set with `options()` don't get passed to `crew` workers and that we probably need something more sophisticated, but how options work with targets are a bit obscure to me.  Any tips or ideas for creating `geotargets_options_set()` and `geotargets_options_get()` functions?

**Comment by wlandau** [comment, marked as answer]
8ab2305d7d1c0a212549970f613fd84916e64278 now implements `tar_resources_custom_format()` which allows you to send custom environment variables to the functions you define in `tar_format()`. These environment variables reach parallel workers. @noamross, this might be useful for #1232.

I think this approach is appropriate when those settings do not invalidate previous work. In other words, if the right decision is to rerun the target when a setting changes, it would be preferable to avoid `tar_resources_custom_format()` and instead bake the setting's value into the body of a `read` or `write` function.

Anyway, here is an example where it is fine to use `tar_resources_custom_format()`. No matter what serialization method you choose, `readRDS()` can still read the file.

```r
library(targets)
tar_option_set(
  controller = crew::crew_controller_local()
)
tar_target(
  name = target_name,
  command = data.frame(x = 1),
  format = tar_format(
    read = function(path) {
      readRDS(file = path)
    },
    write = function(object, path) {
      version <- as.integer(Sys.getenv("SERIALIZATION", unset = ""))
      saveRDS(object = object, file = path, version = version)
    }
  ),
  resources = tar_resources(
    custom_format = tar_resources_custom_format(
      envvars = c(SERIALIZATION = "3")
    )
  )
)
```

**Reply by Aariq** [reply]
Thanks!  This is great and simplifies things a lot for `geotargets`.
